src/Uno.UI/MixinGeneration.targets

```diff
@@ -110,28 +99,31 @@
 
 	<ItemGroup>
 	  <MixinDefinition Include="$(MSBuildThisFileFullPath)" Condition="$(IsMonoAndroid)">
-		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\Android\BaseActivity.Callbacks.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\Android;Configuration=$(Configuration);Platform=$(Platform)</Properties>
+		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\Android\BaseActivity.Callbacks.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\Android\BaseActivity.Callbacks.cs;Configuration=$(Configuration);Platform=$(Platform)</Properties>
 	  </MixinDefinition>
 	  <MixinDefinition Include="$(MSBuildThisFileFullPath)" Condition="$(IsMonoAndroid)">
-		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\Android\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\Android;Configuration=$(Configuration);Platform=$(Platform)</Properties>
+		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\Android\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\Android\FrameworkElementMixins.cs;Configuration=$(Configuration);Platform=$(Platform)</Properties>
 	  </MixinDefinition>
 	  <MixinDefinition Include="$(MSBuildThisFileFullPath)">
-		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\DependencyPropertyMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins;Configuration=$(Configuration);Platform=$(Platform)</Properties>
+		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\DependencyPropertyMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\DependencyPropertyMixins.cs;Configuration=$(Configuration);Platform=$(Platform)</Properties>
 	  </MixinDefinition>
 	  <MixinDefinition Include="$(MSBuildThisFileFullPath)" Condition="$(IsXamarinIOS)">
-		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\iOS\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\iOS;Configuration=$(Configuration);Platform=$(Platform)</Properties>
+		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\iOS\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\iOS\FrameworkElementMixins.cs;Configuration=$(Configuration);Platform=$(Platform)</Properties>
 	  </MixinDefinition>
 	  <MixinDefinition Include="$(MSBuildThisFileFullPath)" Condition="$(IsXamarinMac)">
-		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\macOS\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\macOS;Configuration=$(Configuration);Platform=$(Platform)</Properties>
+		<Properties>InputFile=$(MSBuildThisFileDirectory)Mixins\macOS\FrameworkElementMixins.tt;OutputPath=$(MSBuildThisFileDirectory)Mixins\macOS\FrameworkElementMixins.cs;Configuration=$(Configuration);Platform=$(Platform)</Properties>
 	  </MixinDefinition>
 	</ItemGroup>
 
+
 	<Message Text="Generating mixins" Importance="high" />
 
+	<Exec Command="dotnet tool install dotnet-t4 --version 2.3.0-preview-0041-g0bb7f1d477 --tool-path dotnet-t4" IgnoreExitCode="true" WorkingDirectory="$(MSBuildThisFileDirectory)..\" />
```

Comment:
IntermediatePath did not work? Also, an actual error should be reported. You should be able to use `Condition="exists(..\dotnet-t4\t4)"` to avoid "already install" errors.

Reply:
I haven't tried yet. This suggestion is not related to the failure in WinUI builds right?

Reply:
For some reason, both `BaseIntermediateOutputPath` and `IntermediateOutputPath` are evaluated as empty when the target is run.

![image](https://user-images.githubusercontent.com/31348972/192218577-370b71cc-47fa-470f-8e27-7e43f992ea8a.png)

Reply:
> You should be able to use Condition="exists(..\dotnet-t4\t4)" to avoid "already install" errors.

This won't work well if the developer has installed the tool as global tool I think. The t4 executable will not be found in the expected directory, though `dotnet tool install` can still fail because the tool is installed already.

Reply:
> This won't work well if the developer has installed the tool as global tool I think. The t4 executable will not be found in the expected directory, though `dotnet tool install` can still fail because the tool is installed already.

Local install is not impacted by a global install, the installer validates for the specified path only.

> For some reason, both `BaseIntermediateOutputPath` and `IntermediateOutputPath` are evaluated as empty when the target is run.

Then let's use `obj` hard-coded (we won't change the path any time soon anyways). I'd rather have this as an artifact of the local build than a top level dependency (for now at least).

Reply:
I figured out why `IntermediateOutputPath` didn't work. The `_InnerGenerateMixins` target is run by `MSBuild` task, and for each MixinDefinition, we need to explicitly pass `IntermediateOutputPath`.

So remaining thing here is WinUI failure if it's still present with the latest changes.